packages/@aws-cdk/aws-certificatemanager/lib/certificate.ts

```diff
@@ -135,3 +148,84 @@ class ImportedCertificate extends Construct implements ICertificate {
     return this.props;
   }
 }
+
+/**
+ * A certificate managed by AWS Certificate Manager.  Will be automatically
+ * validated using DNS validation against the specified Route 53 hosted zone.
+ */
+export class DNSValidatedCertificate extends Construct implements ICertificate  {
+  public readonly certificateArn: string;
+  private normalizedZoneName: string;
+  private hostedZoneId: string;
+  private domainName: string;
+
+  constructor(scope: Construct, id: string, props: DNSValidatedCertificateProps) {
+    super(scope, id);
+
+    this.domainName = props.domainName;
+    this.normalizedZoneName = props.hostedZone.zoneName;
+    // Remove trailing `.` from zone name
+    /* istanbul ignore if */
+    if (this.normalizedZoneName.endsWith('.')) {
+      this.normalizedZoneName = this.normalizedZoneName.substring(0, this.normalizedZoneName.length - 1);
+    }
+
+    // Remove any `/hostedzone/` prefix from the Hosted Zone ID
+    this.hostedZoneId = props.hostedZone.hostedZoneId;
+    /* istanbul ignore if */
+    if (this.hostedZoneId.startsWith('/hostedzone/')) {
+      this.hostedZoneId = this.hostedZoneId.split('/hostedzone/')[1];
```

Comment:
Technically this doesn't do what you said it would do if the string `"/hostedzone/"` occurs more than once in your string.

How about a regex?

```ts
this.hostedZoneId = this.hostedZoneId.replace(/^\/hostedzone\//, '');
```

Reply:
Good idea, thanks!